Client.xcodeproj/project.pbxproj

```diff
@@ -4592,7 +4592,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "# Type a script or drag a script file from your workspace to insert its path.\nbash $PWD/sdk_generator.sh\n";
+			shellScript = "# Type a script or drag a script file from your workspace to insert its path.\nbash $PWD/sdk_generator.sh -m $SRCROOT/docs\n";
```

Comment:
I don't think these can be combined in a single step.  To be be 100% correct as a build step, `sdk_generator.sh` should only run when input files have changed, or output files are missing.

Given that autogenerated .swift files are not to be committed -which we consider a best practice- the script will run on a fresh checkout, but now it will modify a committed file in docs, and the problem exists of a locally modified file that was not intentionally modified.

I think this will work as a separate build step. This step should *only* operate for the documentation, and the input file is `$(SRCROOT)/Client/metrics.yaml` and the output file is the doc file.

Reply:
The docs files that are regenerated in this case should not have any changes in them and should be identical to the docs generated by the original commit that added them.  Granted, they may have a new "modified" file date, but I don't think that is enough to show up as a difference to git.

We still think the doc generation step can be separated if this is still a concern, it will just take a slight modification to the sdk_generator.sh script.

Reply:
sgtm, if it doesn't show up as modified locally, it is ok as-is.